pkg/option/config.go

```diff
@@ -3731,6 +3731,11 @@ func (c *DaemonConfig) checkIPAMDelegatedPlugin() error {
 		if c.EnableEndpointHealthChecking {
 			return fmt.Errorf("--%s must be disabled with --%s=%s", EnableEndpointHealthChecking, IPAM, ipamOption.IPAMDelegatedPlugin)
 		}
+		// Ingress controller requires cilium-agent to create an IP address specifically
+		// for differentiating ingress traffic, which is not possible with delegated IPAM
+		if c.EnableIngressController {
+			return fmt.Errorf("--%s must be disabled with --%s=%s", EnableIngressController, IPAM, ipamOption.IPAMDelegatedPlugin)
+		}
```

Comment:
should we also block `c.EnableEnvoyConfig`? I see that Cilium checks that flag when deciding whether to allocate an ingress IP here: https://github.com/cilium/cilium/blob/f3c921be6552049e0526bf4de15855aa4399836f/daemon/cmd/ipam.go#L312

(From [these docs](https://docs.cilium.io/en/stable/network/servicemesh/ingress/), I believe EnableIngressController implies EnableEnvoyConfig, but not the other way around.)

Reply:
This is correct, if envoy is the issue here we should block on that as it is used in many more cases than just ingress (gateway api/ l7 proxy)